shell/browser/ui/message_box_gtk.cc

```diff
@@ -38,10 +42,14 @@ MessageBoxSettings::~MessageBoxSettings() = default;
 
 namespace {
 
+// <ID, messageBox> map
+base::NoDestructor<std::map<int, GtkWidget*>> g_dialogs;
```

Comment:
All the usages in Chromium of `base::NoDestructor` wrap it inside a function like:
```
const std::string& GetLineSeparator() {
 // Forwards to std::string(size_t, char, const Allocator&) constructor.
  static const base::NoDestructor<std::string> s(5, '-');
  return *s;
}
```

Is it OK to have it at the global level like this...?

Reply:
Just found the doc says 

```
// Note that since the destructor is never run, this *will* leak memory if used
// as a stack or member variable. Furthermore, a NoDestructor<T> should never
// have global scope as that may require a static initializer.
```

so yeah they must be wrapped in a function.